src/phoenix/server/api/subscriptions.py

```diff
@@ -245,16 +257,49 @@ def to_openai_chat_completion_param(
                 }
             )
         if role is ChatCompletionMessageRole.AI:
-            return ChatCompletionAssistantMessageParam(
-                {
-                    "content": content,
-                    "role": "assistant",
-                }
-            )
+            if tool_calls is None:
+                return ChatCompletionAssistantMessageParam(
+                    {
+                        "content": content,
+                        "role": "assistant",
+                    }
+                )
+            else:
+                return ChatCompletionAssistantMessageParam(
+                    {
+                        "content": content,
+                        "role": "assistant",
+                        "tool_calls": self.to_openai_tool_call_param(tool_calls),
+                    }
+                )
         if role is ChatCompletionMessageRole.TOOL:
-            raise NotImplementedError
+            if tool_call_id is None:
+                raise ValueError("tool_call_id is required for tool messages")
+        return ChatCompletionToolMessageParam(
+            {"content": content, "role": "tool", "tool_call_id": tool_call_id}
+        )
         assert_never(role)
 
+    def to_openai_tool_call_param(
+        self,
+        tool_calls: List[JSONScalarType],
+    ) -> List["ChatCompletionMessageToolCallParam"]:
+        from openai.types.chat import ChatCompletionMessageToolCallParam
+
+        return [
+            ChatCompletionMessageToolCallParam(
+                id=tool_call.get("id", ""),
+                function={
+                    "name": tool_call.get("function", {}).get("name", ""),
+                    "arguments": safe_json_dumps(
+                        tool_call.get("function", {}).get("arguments", "")
+                    ),
+                },
+                type="function",
+            )
+            for tool_call in tool_calls
+        ]
```

Comment:
I would make this input a single tool call and return a single tool call param for the sake of granularity

Reply:
ahh sounds good